src/Storages/IStorage.h

```diff
@@ -572,6 +572,11 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
         throw Exception(ErrorCodes::NOT_IMPLEMENTED, "Part moves between shards are not supported by storage {}", getName());
     }
 
+    virtual void pushDependencies()
```

Comment:
Push where ? Please add comments

Reply:
Well ... it is intentionally bad naming.
I think it should be more generic, preSetup() or something like this.